lighthouse-core/lib/i18n.js

```diff
@@ -66,6 +67,21 @@ const formats = {
   },
 };
 
+/**
+ * Look up the best available locale for the requested language, falling back by
+ * looking at progressively shorter prefixes (`de-CH-1996` -> `de-CH` -> `de`),
+ * and finally returning the default locale ('en-US') if no match is found.
+ * @param {string=} locale
+ * @return {LH.Locale}
+ */
+function lookupLocale(locale = MessageFormat.defaultLocale) {
+  // TODO: could do more work to sniff out defaultLocale
+  const canonicalLocale = Intl.getCanonicalLocales(locale)[0];
```

Comment:
compat-wise, how do we look here across browsers and node. do we need this one conditionally polyfilled too?

Reply:
> compat-wise, how do we look here across browsers and node. do we need this one conditionally polyfilled too?

it comes with `Intl`, and it seems like we're assuming at least that exists? The function doesn't check valid tags or need a list of real locales or anything, it's just the capitalization and...maybe some other formatting stuff I can't remember.

Though from https://nodejs.org/api/intl.html it seems like maybe there's no `Intl` object at all if node was compiled with `--with-intl=none`?

Reply:
the mdn compat table is out of date, but it at least indicates support was added later after the Intl basics:

![image](https://user-images.githubusercontent.com/39191/43419371-8e0f3be0-93f5-11e8-81ce-9e8ea9d70592.png)

unfortunately the only polyfill appears to be deep inside of Intl.js so hopefully we don't need it?